### PR TITLE
Feat/css colors helpers

### PR DESCRIPTION
**Pull request by rustem-nasyrov**
Related issue: #1222 
closes https://github.com/epicmaxco/vuestic-ui/issues/764 #2169

image

## Description
Added customized classes for the given colors configuration.
* Created `colorsCustomClasses` plugin for the Vuestic-UI.
* Added description in the docs, for the both translations.
* Added examples of usage.

## Types of changes
- [x] New feature (non-breaking change which adds functionality)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/docs/src/page-configs/getting-started/configuration-guide/code-examples/colors-custom-classes.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export const colorsCustomClasses = `
+// main.js
+import { VuesticPlugin } from 'vuestic-ui'
+import 'vuestic-ui/dist/vuestic-ui.css'
+
+const app = createApp(App)
+
+app.use(VuesticPlugin, {
+  colors: { ... },
+  colorsCustomClasses: [
```
Comment (aluarius) [inline packages/docs/src/page-configs/getting-started/configuration-guide/code-examples/colors-custom-classes.ts:10]: Don't we should keep colorsCustomClasses inside of colors object?

File: packages/ui/src/services/colors-custom-classes/plugin/index.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import type { ColorsCustomClassesConfig, ColorsCustomClassesConfigItem } from '../types'
+
+import { ColorsCustomClassesPresets } from '../config/default'
+
+export const getColorsCustomClassesDefaultConfig = () => ColorsCustomClassesPresets
+
+const getColorsCustomClassesHelpers = (helpers: ColorsCustomClassesConfig, colors: ColorVariables) => {
+  const colorsEntries = Object.entries(colors)
+
+  return helpers.reduce((acc, helper: ColorsCustomClassesConfigItem) => acc.concat(
```
Comment (aluarius) [inline packages/ui/src/services/colors-custom-classes/plugin/index.ts:18]: Very hard to read this part.

**Review by aluarius — APPROVED** [review]

File: packages/ui/src/services/colors-custom-classes/plugin/index.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    colorsEntries.map(([colorName, colorValue]) => ({
+      ...helper,
+      postfix: helper.postfix ? helper.postfix : colorName,
+      value: helper.value ? helper.value : colorValue,
```
Comment (m0ksem) [inline packages/ui/src/services/colors-custom-classes/plugin/index.ts:22]:
```suggestion
      postfix: helper.postfix ?? colorName,
      value: helper.value ?? colorValue,
```

File: packages/ui/src/services/colors-custom-classes/plugin/index.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+const getColorsCustomClassesStyles = (helpers: ColorsCustomClassesConfig) =>
+  helpers.reduce((styles, helper) => {
+    const style = [helper.property].flat().map(prop => `${prop}: ${helper.value};`).join('')
+
+    styles += `.va-${helper.prefix}--${helper.postfix} { ${style} }`
+
+    return styles
+  }, '')
```
Comment (m0ksem) [inline packages/ui/src/services/colors-custom-classes/plugin/index.ts:34]: This is not one-liner, so prefer to {} or even function keyword. Can damage brain.
```suggestion
const getColorsCustomClassesStyles = (helpers: ColorsCustomClassesConfig) => {
  return helpers.reduce((styles, helper) => {
      const style = [helper.property].flat().map(prop => `${prop}: ${helper.value};`).join('')
  
      styles += `.va-${helper.prefix}--${helper.postfix} { ${style} }`
  
      return styles
    }, '')
  }
```

File: packages/ui/src/services/colors-custom-classes/plugin/index.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  )
+}
+
+export const ColorsCustomClassesPlugin = defineVuesticPlugin(() => ({
+  install () {
+    if (isServer()) { return }
+
+    const { globalConfig } = useGlobalConfig()
+
+    watch(() => globalConfig.value.colorsCustomClasses as ColorsCustomClassesConfig, (newHelpers: ColorsCustomClassesConfig) => {
```
Comment (m0ksem) [inline packages/ui/src/services/colors-custom-classes/plugin/index.ts:51]: Prefer not to use `as`.

File: packages/ui/src/services/global-config/global-config.ts [hunk 6]
```diff
@@ -19,6 +20,7 @@ export const createGlobalConfig = () => {
     components: getComponentsDefaultConfig(),
     breakpoint: getBreakpointDefaultConfig(),
     i18n: getI18nConfigDefaults(),
+    colorsCustomClasses: getColorsCustomClassesDefaultConfig(),
```
Comment (m0ksem) [inline packages/ui/src/services/global-config/global-config.ts:23]: `colorClasses` ? What `custom` stands for?

**Review by m0ksem — APPROVED** [review]
We might mention tailwind integration somewhere here.

File: packages/docs/src/page-configs/getting-started/configuration-guide/code-examples/colors-custom-classes.ts (hunk 1 above)
Comment (rustem-nasyrov) [reply packages/docs/src/page-configs/getting-started/configuration-guide/code-examples/colors-custom-classes.ts]: It depends on it, but `colorsClasses` is separate service.

**Review by aluarius — APPROVED** [review]

**Review by m0ksem — APPROVED** [review]

**Pull request merged**